frontend/src/metabase/visualizations/click-actions/Mode/Mode.ts

```diff
@@ -58,6 +58,7 @@ export class Mode {
         clicked?.dimensions,
       );
 
+      // TODO: those calculations are really expensive and should be memoized at some level
```

Comment:
@deniskaber @alxnddr any idea how to speed this up?

Reply:
Can we make them lazy and avoid precomputing?

Reply:
current logic assumes onClick handler is available before rendering, I would try to make it lighter and maybe cache on some component level